plancheck/expect_deferred_change_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package plancheck_test
+
+import (
+	"context"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	r "github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/internal/testing/testprovider"
+	"github.com/hashicorp/terraform-plugin-testing/internal/testing/testsdk/providerserver"
+	"github.com/hashicorp/terraform-plugin-testing/internal/testing/testsdk/resource"
+	"github.com/hashicorp/terraform-plugin-testing/plancheck"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func Test_ExpectDeferredChange_Reason_Match(t *testing.T) {
+	t.Parallel()
+
+	r.Test(t, r.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_9_0),
+			tfversion.SkipIfNotPrerelease(),
```

Comment:
I'm going to introduce a change to our terraform repo to ensure this CI actually has an alpha version to run these 😆

Reply:
https://github.com/hashicorp/terraform-devex-repos/pull/71

Reply:
Added, `terraform-plugin-testing` tests are running on `1.9.0-alpha20240501` - 
https://github.com/hashicorp/terraform-plugin-testing/actions/runs/8944501602/job/24635889715